app/assets/javascripts/courseEdit/components/ChangeParentModal.js

```diff
@@ -0,0 +1,161 @@
+import _ from 'underscore'
+import React, { PropTypes } from 'react'
+import Modal from 'react-bootstrap-modal'
+
+import request from '../../shared/requests/request'
+
+import { APIRoutes } from '../../shared/routes'
+import { Images } from '../../utils/helpers/image_helpers'
+
+import SimpleModal from '../../shared/components/widgets/SimpleModal'
+
+class ChangeParentModal extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      sections: this.props.course.sections,
+      subsections: this.getSubsections(this.props.course.sections, this.props.selectedSection),
+      selectedSection: this.props.selectedSection,
+      selectedSubsection: this.props.selectedSubsection,
+    }
+
+    this.handleSectionSelect = this.handleSectionSelect.bind(this)
+    this.handleSubsectionSelect = this.handleSubsectionSelect.bind(this)
+  }
+
+  getSubsections(sections, sectionId) {
+    const section = _.find(sections, function(section) { return section.id == sectionId })
+
+    return section != null ? section.subsections : []
+  }
+
+  updateSubsections(sectionId) {
+    const subsections = this.getSubsections(this.state.sections, sectionId)
+
+    this.setState({ subsections: subsections })
+  }
+
+  getDropdownStyle() {
+    return {
+      backgroundImage: `url(${Images.dropdown_arrow})`,
+    }
+  }
+
+  handleSectionSelect(e) {
+    if (e.target.value != -1) {
+      this.updateSubsections(e.target.value)
+      this.setState({ selectedSection: e.target.value })
+    } else {
+      this.setState({
+        selectedSection: -1,
```

Comment:
you might wanna have some constant like `SECTION_NOT_SELECTED` or something instead of hardcoding `-1` every time so its harder to make a mistake/ have to worry about typing the right number later on